daemon/daemon.go

```diff
@@ -407,20 +407,12 @@ func (daemon *Daemon) reserveName(id, name string) (string, error) {
 
 		conflictingContainer, err := daemon.GetByName(name)
 		if err != nil {
-			if strings.Contains(err.Error(), "Could not find entity") {
-				return "", err
-			}
```

Comment:
I agree that we should probably separate containers by graphdriver. Other way would be to load containers into a broken state and allow users to delete, but I think this will be more confusing to the user with benefits only in very rare cases. Magically removing containers on conflict seems like the worst option.